### DISCUSSION TITLE
useFieldArray and type-check issues

### DISCUSSION THREAD

**Discussion by gregordotjs** [react-hook-form/react-hook-form#5698] [Q&A]
I updated to the latest version of RHF and I got a couple of TypeScript issues. Namely, this part:

```javascript
{fields.map((field, i: number) => (
  <tr key={field.id}>
    <td>
      <Controller
        name={`Config.${i}.Disabled` as const}
        render={({ field }) => (
          <Checkbox
            {...field}
            label={`Disable`}
          />
        )}
      />
    </td>
```

Produces a really odd error: 			
> Type '`` `Config.${number}.Disabled` ``' is not assignable to type '`` `${string}` | `${string}.${string}` `` | `` `${string}.${number}` ``'.
  Type '`` `Config.${number}.Disabled` ``' is not assignable to type '`` `${string}.${number}` ``'.
> 260                 name={`` `Config.${i}.Disabled` `` as const}

The same happens without "as const".

I get similar errors within `fields.map` part with "getValues(`` `config.${i}.ids` ``)", but it works with "field[`` `config.${i}.ids` ``]"

Things work, though, it's just this error that gets always thrown. Any ideas what's going on?

**Comment by bluebill1049** [comment]
can you provide a codesandbox?
